Align FrozenDate API docs with chronos

markstory opened the pull request:
Align the documentation with what is in Chronos. Chronos was changed in https://github.com/cakephp/chronos/pull/227 (a820c1b1a523df341468d536ca7a2900ccaabc05) to fix problems with timezones that are far away from UTC.

Fixes #15418

othercorey commented:
Should we just inherit the chronos docblock?

ADmad commented on src/I18n/FrozenDate.php:
```diff
@@ -124,7 +124,7 @@ class FrozenDate extends ChronosDate implements I18nDateTimeInterface
      * Date instances lack time components, however due to limitations in PHP's
      * internal Datetime object the time will always be set to 00:00:00, and the
-     * timezone will always be UTC. Normalizing the timezone allows for
+     * timezone will always be server local time. Normalizing the timezone allows for
```
The statement "Normalizing the timezone.." doesn't seem necessary now.

markstory commented:
> Should we just inherit the chronos docblock?

Do docblocks inherit across projects?

othercorey commented:
Have to use the same parameter name as Chronos.

othercorey committed:
Align FrozenDate API docs with chronos

Align the documentation with what is in Chronos. Chronos was changed in
cakephp/chronos#227 (a820c1b1a523df341468d536ca7a2900ccaabc05)
to fix problems with timezones that are far away from UTC.

othercorey commented:
I reverted the fix back to wording change to avoid issues with Chronos.
